### DISCUSSION TITLE
Android Auto Support

### DISCUSSION THREAD

**Discussion by BradNut** [FossifyOrg/General-Discussion#581] [Ideas]
This music player is my go to for a dedicated player. It would be nice to have Android Auto support at some point. Just wondering if anyone else wants this feature to be implemented at some point. Thank you!

**Comment by hades** [comment]
It looks like Android Auto support was intentionally disabled here:

https://github.com/FossifyOrg/Music-Player/blame/60688c8f378b6fc75875181fb4142f31c1132f77/app/src/main/AndroidManifest.xml#L44

Is there any work necessary before this can be re-enabled? If so, I'm willing to volunteer to help with that.

**Reply by Aga-C** [reply]
Here you can find why it was disabled: https://github.com/SimpleMobileTools/Simple-Music-Player/pull/603#issuecomment-1817365609 (and for the record, here's PR that was originally adding Android Auto: https://github.com/SimpleMobileTools/Simple-Music-Player/pull/587).

If that's not enough, maybe @naveensingh could tell you more what needs to be done in the app to revert it in a way that we won't have any problems with Google Play. And most importantly, would we like to implement it now.

**Reply by hades** [reply]
Is the app even on Play Store? I couldn't find it, and it's not mentioned in the README.

Either way, I can help fixing any problems that prevent Android Auto support, or actually anything preventing from publishing in Google Play for that matter.

**Reply by inson1** [reply]
@hades not yet, it will be
https://github.com/orgs/FossifyOrg/discussions/57#discussioncomment-8145586

**Comment by hades** [comment]
@naveensingh it is mentioned in [1] that Google Play requires search capability for Android Auto.

I've tinkered with the code a bit, and this works as a rudimentary search implementation: https://github.com/FossifyOrg/Music-Player/compare/master...hades:Music-Player:aa-support-wip

Is this in the ballpark of what's needed to be implemented?

[1] https://github.com/SimpleMobileTools/Simple-Music-Player/pull/603#issuecomment-1817365609

**Reply by naveensingh** [reply]
> Is this in the ballpark of what's needed to be implemented?

Yep, that should do it. Some information is here: https://github.com/androidx/media/issues/204

**Comment by muesli79** [comment]
Android Auto would be great!

**Comment by SinnerK0N** [comment]
I made a fork with Android Auto support enabled that im gonna keep up-to-date with upstream
https://github.com/SinnerK0N/Music-Player
more details in the readme if anyone is interested

**Reply by SinnerK0N** [reply]
https://github.com/FossifyOrg/Music-Player/assets/64489962/2f4e68f8-3c0a-4ff0-8f31-37291a8dc7f7

**Reply by GavinBrelstaff** [reply]
If you are sideloading SinnerK0N's mod you'll need to use developer mode in Android Auto as described here: https://www.reddit.com/r/AndroidAuto/comments/10k0uyc/missing_apps_in_android_auto_launcher_menu/

**Comment by dl200010** [comment]
I definitely would love this! This is basically the only thing keeping from switching over to this app.

**Comment by BradNut** [comment]
Any chance of getting this into the main Fossify version?

@SinnerK0N would you provide a pull request into this repository?

**Reply by dl200010** [reply]
@hades already has sent a pull request with this and a search function.

**Reply by hades** [reply]
To be clear: I didn't send a pull request, I just asked if this was what Google Play wants in order for Android Auto to be enabled. I didn't get any response to this.

**Reply by naveensingh** [reply]
> I didn't get any response to this.

I missed the mentions somehow. I replied to your comment.

**Reply by dl200010** [reply]
@hades Sorry, man. I thought it was a pull request. I was mistaken.

**Comment by naveensingh** [comment]
Closing in favor of https://github.com/FossifyOrg/Music-Player/issues/172
